Cpp2IL.Core/Model/Contexts/ArrayTypeAnalysisContext.cs

```diff
@@ -19,5 +19,7 @@ public ArrayTypeAnalysisContext(Il2CppType rawType, AssemblyAnalysisContext refe
 
     public override string DefaultName => $"{ElementType.Name}[{Rank}]";
 
+    public sealed override bool IsValueType => false;
```

Comment:
I made the overrides sealed, but I feel like maybe these classes should be sealed instead. I can't think of a valid reason for someone to derive from them.

Reply:
Either is probably fine. For now I think leaving the override as sealed will suffice. More flexibility isn't a bad thing.